### PR TITLE
Duplication in PMD reporter is eliminated; additions to PMD reporter tests

### PR DESCRIPTION
**Pull request by Uranium-92**
This is a minor refactoring that is needed to implement reporting compilation errors and warnings in PMD reporter (https://github.com/oclint/oclint/issues/398).

### CODE REVIEW

**Comment by Uranium-92** [comment]
@ryuichis, can you merge this if it is OK, please? I will open new pull request with solution to https://github.com/oclint/oclint/issues/398 once this one is merged.

**Comment by ryuichis** [comment]
@Uranium-92 Thanks for working on this. I promise I will take a look at this no later than Tuesday.

File: oclint-reporters/reporters/PMDReporter.cpp [hunk 1]
```diff
@@ -29,6 +47,29 @@ class PMDReporter : public Reporter
...
+    {
+        out << "<file name=\"" << violation.path << "\">" << std::endl;
+        out << "<violation ";
+        out << "begincolumn=\"" << violation.startColumn << "\" ";
+        out << "endcolumn=\"" << violation.endColumn << "\" ";
+        out << "beginline=\"" << violation.startLine << "\" ";
+        out << "endline=\"" << violation.endLine << "\" ";
+        if (ruleDescription)
+        {
+            out << "priority=\"" << ruleDescription->priority << "\" ";
```
Comment (ryuichis) [inline oclint-reporters/reporters/PMDReporter.cpp]: Wondering if it's a good idea to add a method `writeViolation(std::ostream &out)` in `PMDRuleDescription` class, which contains this line and below two lines. And here, we can basically have `ruleDescription.writeViolation(out)`.

Comment (Uranium-92) [inline oclint-reporters/reporters/PMDReporter.cpp]: Good point. Thanks! Will include fix for that in the next pull request ;)

**Pull request merged**